pilot/proxy/envoy/watcher.go

```diff
@@ -55,23 +55,25 @@ type CertSource struct {
 }
 
 type watcher struct {
-	agent    proxy.Agent
-	role     model.Node
-	config   meshconfig.ProxyConfig
-	certs    []CertSource
-	pilotSAN []string
+	agent      proxy.Agent
+	role       model.Node
+	config     meshconfig.ProxyConfig
+	certs      []CertSource
+	pilotSAN   []string
+	enableAuth bool
 }
 
 // NewWatcher creates a new watcher instance from a proxy agent and a set of monitored certificate paths
 // (directories with files in them)
 func NewWatcher(config meshconfig.ProxyConfig, agent proxy.Agent, role model.Node,
-	certs []CertSource, pilotSAN []string) Watcher {
+	certs []CertSource, pilotSAN []string, enableAuth bool) Watcher {
```

Comment:
I think breaks ingress cert renewal?

Reply:
That's a good point. I think we may need to differentiate the file path as well ...

Reply:
I will move the logic out the the cmd main function.